WiseBuy/View Controllers/DealsViewController.m

```diff
@@ -30,28 +31,48 @@ -(void)viewDidLoad {
 }
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:YES];
-    NSLog(@"%@", self.barcode);
-//    [APIManager fetchDealsFromUPCDatabase:@"53039031"];
 
-    if (![DatabaseManager checkIfItemAlreadyExist:@"53039031"]) {
-        [APIManager fetchDealsFromEbayAPI:@"53039031"];
+    if (![DatabaseManager checkIfItemAlreadyExist:self.barcode]) {
+        [APIManager fetchDealsFromEbayAPI:self.barcode];
         [APIManager fetchDealsFromUPCDatabase:self.barcode];
         [APIManager fetchDealsFromSearchUPCAPI:self.barcode];
     }
+    JGProgressHUD *HUD = [JGProgressHUD progressHUDWithStyle:JGProgressHUDStyleLight];
+    HUD.textLabel.text = @"Waiting for deals to be displayed";
+    [HUD showInView:self.view];
     
```

Comment:
Any reason for not incorporating this in `setLoadingState` and storing the HUD as a local var? This would allow you to incapsulate all of the loading code into a single function instead of having it segmented between `viewWillAppear` and `setLoadingState`

Reply:
i see and will fix this